custom-script/utils.py

```diff
@@ -18,7 +18,7 @@
 MODEL_INPUTS_STR = ", ".join(MODEL_INPUTS)
 
 
-def prepare_data(eopatches: np.ndarray, train_eopatches: np.ndarray):
+def prepare_data(eopatches: np.ndarray, train_eopatches: np.ndarray) -> Tuple[np.ndarray]:
```

Comment:
Why `eopatches: np.ndarray` and not something much more informative like `eopatches: List[EOPatch]`? Similar for `train_eopatches`: why `np.ndarray` if it is (or could be) just a list of boolean values?

Reply:
The question is WHY would you cast it into an numpy array. EOPatches are not efficiently stored in numpy arrays, so it doesn't benefit you much. A quick look at the code shows that it is mainly used so that you can use `eopatches[train_eopatches]` in lines after 26. But the same effect could be achieved with a some clever use of list comprehension. Perhaps even a simple for loop, since it looks like you're iterating over `eopatches` 6 times when building the arrays.

And your return type is wrong, `Tuple[np.ndarray]` defines a 1-element tuple, while you return a 4-dimensional tuple. You need to use `Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]`. One can use `Tuple[np.ndarray, ...]` to signify any-dimension tuples of ndarrays, but that is not your case.

You should also consider improving the code. You return a 4-tuple of numpy arrays, that is very very easy to mix up. Why not return a dictionary with meaningfuly names instead?

Reply:
done